packages/beachball/src/publish/toposortPackages.ts

```diff
@@ -0,0 +1,46 @@
+import * as _ from 'lodash';
+import toposort from 'toposort';
+import { PackageInfos } from '../types/PackageInfo';
+
+/**
+ * Topological sort the packages based on its dependency graph.
+ * Dependency comes first before dependent.
+ * @param packages Packages to be sorted.
+ * @param packageInfos PackagesInfos for the sorted packages.
+ */
+export function toposortPackages(packages: string[], packageInfos: PackageInfos): string[] {
+  const packageSet = new Set(packages);
+  const dependencyGraph: [string | undefined, string][] = [];
+
+  packages.forEach(pkgName => {
+    let allDeps: string[] = [];
```

Comment:
Possibly important in bigger repos - but you can use a Set to get uniques without having to deal with using lodash for it... Sets are also iterable so you shouldn't have to convert it back to an array